system/include/emscripten/val.h

```diff
@@ -575,15 +575,34 @@ namespace emscripten {
         };
     }
 
-    template<typename T>
-    std::vector<T> vecFromJSArray(val v) {
-        auto l = v["length"].as<unsigned>();
+    template <typename T>
+    std::vector<T> vecFromJSArray(const val & v) {
+        const size_t l = v["length"].as<size_t>();
 
         std::vector<T> rv;
-        for(unsigned i = 0; i < l; ++i) {
+        rv.reserve(l);
+        for (size_t i = 0; i < l; ++i) {
             rv.push_back(v[i].as<T>());
         }
 
         return rv;
-    };
+    }
+
+    template <typename T>
+    std::vector<T> convertJSArrayToNumberVector(const val & v) {
+        const size_t l = v["length"].as<size_t>();
+
+        std::vector<T> rv;
+        rv.resize(l);
+
+        // Copy the array into our vector through the use of typed arrays.
+        // The main issue however, is that if it contains something else than numbers (for example a string),
+        // it will try to convert it as if using Number().
```

Comment:
seems harmless... does what the function name says it will do.   not sure  why this is an issue, but the link to the emca docs is nice.

Reply:
Ah right, I initially put this comment here because the function was a specialization of `vecFromJSArray`. I can change this